### DISCUSSION TITLE
What is the purpose of tsconfig file if typescript was not selected while project setup?

### DISCUSSION THREAD

**Discussion by korywka** [sveltejs/kit#2809] [Q&A] [answered]
The same question is about `global.d.ts`.

Thanks.

**Comment by bluwy** [comment, marked as answer]
`tsconfig.json` and `global.d.ts` are used by your IDE (e.g. vscode) to provide intellisense/autocompletion for your project. The main usecase in SvelteKit is to set up the path alias so that auto imports take `$lib` into account too. All these doesn't necessarily need typescript installed in your project because your IDE would likely have a copy built-in already.

**Reply by korywka** [reply]
@bluwy Thanks. I think this part worth to be explained in documentation. This is the difference between react create app, next.js and svelte kit.

**Reply by bluwy** [reply]
Perhaps we can have some explanation under the default scaffold project readme, but I'm not too convinced if it's needed. The end-user shouldn't be removing the default files unless they know what they're doing. This discussion can serve as the explanation if they're curious though.
